Pull Request for Comments to the Syllabus (#45)
robotframework/robotframework-RFCP-syllabus: website/docs/chapter-02/02_suitefile_syntax.md

**Example 2**: In the `*** Settings ***` section, the settings are separated from their values by four or more spaces.

```robotframework
*** Settings ***
# SETTINGS HDR |
Documentation     This is the first line of documentation.
#  SETTING  | SEP |                 VALUE                | EOL
...   # just CONTINUATION and End Of Line
...            This is the second line of documentation.
# CONTINUATION |                VALUE                  | EOL
Resource     keywords.resource
# SET  | SEP |     VALUE     | EOL
```

All elements themselves in their section are written without any indentation.
So settings in the `*** Settings ***` section, test cases in the `*** Test Cases ***` section,
and keywords in the `*** Keywords ***` section are written without any indentation.
However, when defining tests|tasks and keywords, indentation is used to define their body, while their name is still un-indented.
So after i.e. a test case name, all subsequent lines that are part of the test case body are indented by two or more spaces.

That means that a body statement always starts with a separator token, followed by a data token, like i.e. variable or keyword as seen in the examples above.

> The body ends when either a new un-indented test case name is defined
> or another section like `*** Keywords ***` starts
> or the end of the file is reached.

Reviewer: ```suggestion
The body ends when either a new un-indented name of the section type is defined (keyword or testcase/task)
or another section like `*** Keywords ***` starts
or the end of the file is reached.
```

By default, each statement is terminated by a line break, allowing only one statement per literal line.
However, for better readability, or to add line breaks in documentation,
statements can span multiple lines by using `...` (three dots) and a separator at the start of the next line with the same indentation level as the line being continued.

A line continuation can only be performed where a separator is expected, like between a keyword name and its arguments or between two arguments or between a setting and its value(s).
In the following example the two keyword calls are logically identical, even though the second one is split over three literal lines.

In documentation settings, line breaks with continuation are interpreted as a line break character.
In Robot Framework documentation syntax, a single line break is treated as a space after interpretation,
whereas two consecutive line breaks are considered a paragraph break.
This allows you to structure documentation in a more readable and organized manner.

**Example**:

```robotframework
*** Settings ***
Documentation     This is the first line of suite documentation.
...

> ...            This is the second line of suite documentation.

Reviewer: ```suggestion
...            This is the first line in the second paragraph of suite documentation.
```

Resource     keywords.resource

*** Test Cases ***
Test Case Name
      [Documentation]    This is the first line of test documentation.
      ...

>       ...                This is the second line of test documentation.

Reviewer: ```suggestion
      ...                This is the first line in the second paragraph of test documentation.
```

      Keyword Call     argument one     argument two
      Keyword Call
      ...            argument one
      ...            argument two
      ${variable_assignment}     Keyword Getter Call
```

## 2.2.3 In-line Comments

::::lo[Learning Objectives]

:::K3[LO-2.2.3]

Be able to add in-line comments to suites.

:::

::::

In Robot Framework comments can be added to lines after the content
by starting the comment with a separator (multiple spaces) and a hash `#`.
The hash `#` is used to indicate that the rest of the line is a comment and is ignored by Robot Framework.

> Same works at the very start of a line, which makes the whole line a comment.

Reviewer 2: This sounds nice if you say "The same works at the very start of a line, which makes the whole line a comment."
Prefix with "The"

Hashes in the middle of a value are considered normal characters and do not need to be escaped.

> If an argument value or any other thing shall start with a hash (`#`)

Reviewer 2: "any other thing" is a bit vague. What constitutes a "thing"?

and it is preceded by a separator (multiple spaces),
the hash must be escaped by a backslash `\` like `Click Element By Css    \#element_id`.

> Block comments are not supported in Robot Framework,

Reviewer 2: Not relevant to this syllabus content, but I would love to see Block comments in Robot Framework or an IDE plugin that really allows you to select text and block comment with multiple #s (but I digress)
